File: 백준/Silver/10815. 숫자 카드/숫자 카드.py
N = input()
cards = list(input().split())
M = input()
nums = list(input().split())

objs = {}

for num in cards:
  objs[num] = True

answer = ["1" if objs.get(num, False) else "0" for num in nums]

print(" ".join(answer))

File: 백준/Silver/10815. 숫자 카드/숫자 카드.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const [N, cards, M, nums] = input;

const objs = {};

cards.split(" ").forEach((item) => (objs[item] = true));

const answer = nums.split(" ").map((item) => (objs[item] ? 1 : 0));

console.log(answer.join(" "));
